File: takeItEasy-presentation/src/main/kotlin/com/sudosoo/takeItEasy/controller/TestController.java
package com.sudosoo.takeItEasy.controller;

import com.sudosoo.takeiteasy.batch.BatchLauncherService;
import com.sudosoo.takeiteasy.dto.post.PostRequestDto;
import com.sudosoo.takeiteasy.dto.post.PostResponseDto;
import com.sudosoo.takeiteasy.redis.RedisService;
import com.sudosoo.takeiteasy.service.PostService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;

import java.util.List;

@RestController
@RequestMapping("/test")
@RequiredArgsConstructor
public class TestController {
    private final BatchLauncherService batchLauncherService;
    private final RedisService redisService;
    private final PostService postService;


    @PostMapping(value = "/batchCreateDummyPost" , name = "batchCreateDummyPost")
    public ResponseEntity<Void> batchCreateDummyPost() {
        //batch를 로직 통한 실행
        batchLauncherService.runBatchJob();
        return ResponseEntity.ok().build();
    }

    @PostMapping(value = "/defaultCreateDummyPost" , name = "createDummyPost")
    public ResponseEntity<Void> defaultCreateDummyPost() throws Exception {
        //기존 로직
        batchLauncherService.runBatchJobV2();

        return ResponseEntity.ok().build();
    }

    @GetMapping(value = "/getV" , name = "getV")
    public List<PostResponseDto> redisGetTest()  {
        String methodName = "PostResponseDto";
        return redisService.getValues(methodName);
    }


    @PostMapping(value = "/createPost" , name = "createPost")
    public PostResponseDto createPost(@RequestBody PostRequestDto requestDto)  {
        return postService.redisTest(requestDto);
    }

    @PostMapping(value = "/synchronize" , name = "repositoryRedisSynchronization")
    public void repositoryRedisSynchronization()  {
        redisService.postRepositoryRedisSynchronization();
    }

}

File: takeItEasy-presentation/src/main/kotlin/com/sudosoo/takeItEasy/controller/TestController.kt
package com.sudosoo.takeItEasy.controller

import com.sudosoo.takeItEasy.dto.post.PostRequestDto
import com.sudosoo.takeItEasy.dto.post.TestPostResponseDto
import com.sudosoo.takeItEasy.service.PostService
import com.sudosoo.takeiteasy.batch.BatchLauncherService
import com.sudosoo.takeiteasy.redis.RedisService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/test")
class TestController (
    val batchLauncherService: BatchLauncherService,
    val redisService: RedisService,
    val postService: PostService
    ){

    @PostMapping("/batchCreateDummyPost", name = "batchCreateDummyPost")
    fun batchCreateDummyPost(): ResponseEntity<Void> {
        //batch를 로직 통한 실행
        batchLauncherService.runBatchJob()
        return ResponseEntity.ok().build()
    }

    @PostMapping("/defaultCreateDummyPost", name = "createDummyPost")
    @Throws(Exception::class)
    fun defaultCreateDummyPost(): ResponseEntity<Void> {
        //기존 로직
        batchLauncherService.runBatchJobV2()

        return ResponseEntity.ok().build<Void>()
    }

    @GetMapping("/getV", name = "getV")
    fun redisGetTest(): List<TestPostResponseDto> {
        val methodName = "PostResponseDto"
        return redisService.getValues(methodName)
    }


    @PostMapping("/createPost", name = "createPost")
    fun createPost(@RequestBody requestDto: PostRequestDto): TestPostResponseDto {
        return postService.redisTest(requestDto)
    }

    @PostMapping("/synchronize", name = "repositoryRedisSynchronization")
    fun repositoryRedisSynchronization() {
        redisService.postRepositoryRedisSynchronization()
    }
}